File: 1669-merge-in-between-linked-lists/main.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {

    currNode := list1
    for range(a-1){
        currNode = currNode.Next
    }

    mergeEndNode := currNode.Next
    for range( b - a + 1){
        mergeEndNode = mergeEndNode.Next
    }

    currNode.Next = list2

    endList2 := list2
    for endList2.Next != nil {
        endList2 = endList2.Next
    }

    endList2.Next = mergeEndNode
    return list1
}

File: 1669-merge-in-between-linked-lists/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        
        curr_node = list1
        for _ in range(a - 1):
            curr_node = curr_node.next
        
        merge_end_node = curr_node.next
        for _ in range(b - a + 1):
            merge_end_node = merge_end_node.next
        
        curr_node.next = list2

        end_of_list2 = list2
        while end_of_list2.next:
            end_of_list2 = end_of_list2.next

        end_of_list2.next = merge_end_node

        return list1

        
